lib/stream.js

```diff
@@ -0,0 +1,32 @@
+/**
+ * Safely streams and parses a JSON response
+ * will reject, but keep original response/text if
+ *  - Response is not valid json
+ *  - Response has already been streamed
+ * @param {Response} response Response object returned from fetch
+ */
+export function safeStreamJSON(response) {
+  if (response && response instanceof Response && !response.bodyUsed) {
+    return response.text()
```

Comment:
Curious why you went with `response.text` & `JSON.parse` instead of `response.json`?

Reply:
Response.json calls text, and the json.parse, but if the streamed text is NOT valid json, you lose that text.

`var response = new Response('Thing');
response.json()`

Will result in an error `Uncaught (in promise) SyntaxError: Unexpected token T in JSON at position 0
    at <anonymous>:2:10`

But since the body has already been used, and then text was not returned in the error message, we don't have any more context to that failure.

Reply:
Huh, interesting. Thanks